dfe-analytical-services/explore-education-statistics
# DFE-357 Data API support for FastTrack links

To test:

1. In blob storage, create a container “fasttrack” if it doesn’t already exist.
2. Create a file  in the fasttrack container with name “a8f78ed9-3ec4-4816-ba1a-fd4d9167a82f“
3. Add the following content to the file:

`{"Id":"a8f78ed9-3ec4-4816-ba1a-fd4d9167a82f","Created":"2019-08-16T09:13:42.7741711Z","Query":{"Configuration":{"TableHeadersConfig":{"ColumnGroups":[[{"Label":"cg0.0","Value":"cg0.0"},{"Label":"cg0.1","Value":"cg0.1"}],[{"Label":"cg1.0","Value":"cg1.0"},{"Label":"cg1.1","Value":"cg1.1"}]],"Columns":[{"Label":"c0","Value":"c0"},{"Label":"c1","Value":"c1"}],"RowGroups":[[{"Label":"rg0.0","Value":"rg0.0"},{"Label":"rg0.1","Value":"rg0.1"}],[{"Label":"rg1.0","Value":"rg1.0"},{"Label":"rg1.1","Value":"rg1.1"}]],"Rows":[{"Label":"r0","Value":"r0"},{"Label":"r1","Value":"r1"}]}},"SubjectId":1,"TimePeriod":{"StartYear":2016,"StartCode":"AY","EndYear":2016,"EndCode":"AY"},"Filters":[70],"BoundaryLevel":null,"GeographicLevel":null,"Indicators":[1,23,26],"Country":["E92000001"],"Institution":null,"LocalAuthority":null,"LocalAuthorityDistrict":null,"LocalEnterprisePartnership":null,"MultiAcademyTrust":null,"MayoralCombinedAuthority":null,"OpportunityArea":null,"ParliamentaryConstituency":null,"Region":null,"RscRegion":null,"Sponsor":null,"Ward":null}}`

4. Perform a GET request :

http://localhost:5000/api/fasttrack/a8f78ed9-3ec4-4816-ba1a-fd4d9167a82f

Expected result: JSON object with fields `id, created, fullTable, query`

`fullTable` includes the `subjectMeta` and results 

`query` is the original query from blob file which was used to generate the FastTrack link.
